utils/ProfileProvider.tsx

```diff
@@ -19,11 +19,9 @@ import { useAuth } from './AuthProvider';
 export interface ProfileContextType {
   profileData: Profile | null;
   profileReady: boolean;
-  hasPlot: boolean | null;
   plantsToAdd: Plant[];
   setProfile: (completeProfile: Profile) => Promise<void>; // Now expects full Profile
   loadProfile: () => Promise<void>;
-  setHasPlot: (plotValue: boolean | null) => void;
   setPlantsToAdd: (plants: Plant[]) => void;
```

Comment:
we should prob separate `setPlantsToAdd` and `plantsToAdd` out into a separate context. otherwise, updating setPlantsToAdd would trigger a re-render of the entire app